server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/JmapPreviewContract.scala

```diff
@@ -0,0 +1,108 @@
+import org.apache.james.mailbox.DefaultMailboxes
+import org.apache.james.mailbox.MessageManager.AppendCommand
+import org.apache.james.mailbox.model.{MailboxPath, MessageId}
+import org.apache.james.mime4j.dom.Message
+import org.apache.james.modules.MailboxProbeImpl
+import org.apache.james.utils.{DataProbeImpl, GuiceProbe}
+import org.awaitility.Awaitility
+import org.junit.jupiter.api.{BeforeEach, Test}
+import reactor.core.scala.publisher.SMono
+
+import scala.jdk.OptionConverters.RichOption
+
+class MessageFastViewProjectionProbe @Inject() (messageFastViewProjection: MessageFastViewProjection) extends GuiceProbe {
+  def retrieve(messageId: MessageId): Optional[MessageFastViewPrecomputedProperties] =
+    SMono.fromPublisher(messageFastViewProjection.retrieve(messageId)).blockOption().toJava
+}
+
+class JmapPreviewProbeModule extends AbstractModule {
+  override protected def configure(): Unit = {
+    Multibinder.newSetBinder(binder(), classOf[GuiceProbe])
+      .addBinding()
+      .to(classOf[MessageFastViewProjectionProbe])
+  }
+}
+
+object JmapPreviewContract {
+  private def createTestMessage: Message = Message.Builder
+    .of
+    .setSubject("test")
+    .setSender(ANDRE.asString())
+    .setFrom(ANDRE.asString())
+    .setSubject("World domination \r\n" +
+      " and this is also part of the header")
+    .setBody("testmail", StandardCharsets.UTF_8)
+    .build
+}
+
+trait JmapPreviewContract {
+  private lazy val slowPacedPollInterval = Duration.ofMillis(100)
+  private lazy val calmlyAwait = Awaitility.`with`
+    .pollInterval(slowPacedPollInterval)
+    .and.`with`.pollDelay(slowPacedPollInterval)
+    .await
+  private lazy val awaitAtMostTenSeconds = calmlyAwait.atMost(10, TimeUnit.SECONDS)
+
+  @BeforeEach
+  def setUp(server: GuiceJamesServer): Unit = {
+    server.getProbe(classOf[DataProbeImpl])
+      .fluent
+      .addDomain(DOMAIN.asString)
+      .addUser(BOB.asString, BOB_PASSWORD)
+
+    requestSpecification = baseRequestSpecBuilder(server)
+      .setAuth(authScheme(UserCredential(BOB, BOB_PASSWORD)))
+      .build
+  }
+
+  @Test
+  def jmapPreviewShouldBeWellRemovedWhenDeleteMailbox(server: GuiceJamesServer): Unit = {
+    val mailboxProbe = server.getProbe(classOf[MailboxProbeImpl])
+    mailboxProbe.createMailbox("#private", BOB.asString, DefaultMailboxes.INBOX)
+
+    val messageId = mailboxProbe.appendMessage(BOB.asString, MailboxPath.inbox(BOB), AppendCommand.builder()
+        .build(createTestMessage))
+      .getMessageId
+
+    val messageFastViewProjectionProbe: MessageFastViewProjectionProbe = server.getProbe(classOf[MessageFastViewProjectionProbe])
+    awaitAtMostTenSeconds.until(() => messageFastViewProjectionProbe.retrieve(messageId).isPresent)
+
+    mailboxProbe.deleteMailbox("#private", BOB.asString, DefaultMailboxes.INBOX)
+    awaitAtMostTenSeconds.until(() => messageFastViewProjectionProbe.retrieve(messageId).isEmpty)
+  }
+}
```

Comment:
What happens with the `Expunged` event now?

My guess: there would be 2 Cassandra queries `DELETE FROM message_fast_view_projection`: 1 from `CassandraMessageFastViewProjectionDeletionCallback` and 1 from the `ComputeMessageFastViewProjectionListener`.

Should `ComputeMessageFastViewProjectionListener` not handle the `Expunged` event anymore?

Reply:
Fair concern. Working on sth else now but I will add a test tomorrow with this case and have a look